src/main/java/de/hpi/bpt/argos/api/entityMapping/EntityMappingEndpointImpl.java

```diff
@@ -0,0 +1,242 @@
+    private static final String EVENT_TYPE_ATTRIBUTE_ID_ATTRIBUTE = "EventTypeAttributeId";
+    private static final String ENTITY_TYPE_ATTRIBUTE_ID_ATTRIBUTE = "EntityTypeAttributeId";
+    private static final String TARGET_STATUS_ATTRIBUTE = "TargetStatus";
+
+    private static final String JSON_PARSE_ERROR_MESSAGE = "cannot parse request body to event mapping";
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setup(Service sparkService) {
+        sparkService.post(EntityMappingEndpoint.getCreateEntityMappingBaseUri(), this::createEntityMapping);
+        sparkService.delete(EntityMappingEndpoint.getDeleteEntityMappingBaseUri(), this::deleteEntityMapping);
+        sparkService.put(EntityMappingEndpoint.getEditEntityMappingBaseUri(), this::editEntityMapping);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String createEntityMapping(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        try {
+            JsonObject jsonBody = jsonParser.parse(request.body()).getAsJsonObject();
+            JsonObject jsonMapping = jsonBody.get(JSON_ENTITY_MAPPING_ATTRIBUTE).getAsJsonObject();
+
+            if (jsonMapping == null) {
+                halt(HttpStatusCodes.BAD_REQUEST, "no event mapping given in body");
+            }
+
+            createEventMappingFromJson(jsonMapping);
+
+        } catch (HaltException halt) {
+            LoggerUtilImpl.getInstance().error(logger,
+                    String.format("cannot create event query: %1$s -> %2$s", halt.statusCode(), halt.body()), halt);
+            throw halt;
+        } catch (Exception e) {
+            LoggerUtilImpl.getInstance().error(logger, JSON_PARSE_ERROR_MESSAGE, e);
+            halt(HttpStatusCodes.ERROR, e.getMessage());
+        }
+
+        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
+        return response.body();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String deleteEntityMapping(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long mappingId = getMappingId(request);
+        EventEntityMapping mapping = PersistenceAdapterImpl.getInstance().getEventEntityMapping(mappingId);
+        if (mapping == null) {
+            halt(HttpStatusCodes.BAD_REQUEST, "given mapping was not found");
+        }
+
+        // delete corresponding mapping conditions
+        List<MappingCondition> conditions = PersistenceAdapterImpl.getInstance().getMappingConditions(mappingId);
+        if (!PersistenceAdapterImpl.getInstance().deleteArtifacts(conditions.toArray(new MappingCondition[conditions.size()]))) {
+            halt(HttpStatusCodes.ERROR, "could not delete mapping conditions");
+        }
+
+        String mappingsUri =  EventTypeEndpoint.getEventTypeEntityMappingsUri(mapping.getEventTypeId());

```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'mapping' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)